inst/@sym/repmat.m

```diff
@@ -51,18 +52,20 @@
     print_usage ();
   end
 
-  cmd = { '(A, n, m) = _ins'
-          'if n == 0 or m == 0:'
-          '    return sp.Matrix(n, m, [])'
-          'if A is None or not A.is_Matrix:'
-          '    A = sp.Matrix([A])'
-          'L = [A]*m'
-          'B = sp.Matrix.hstack(*L)'
-          'L = [B]*n'
-          'B = sp.Matrix.vstack(*L)'
-          'return B' };
+  if n == 0 || m == 0
+    [nrows_A, ncols_A] = size (A);
+    B = zeros (n * nrows_A, m * ncols_A);
+    return
+  end
+
+  make_1d_cell = @(n, init) cellfun (@(unused) init, cell (1, n), ...
+                                     'UniformOutput', false);
+  MM = make_1d_cell (m, A);
```

Comment:
I find this a bit hard to read but I don't see an alternative way ...

consider:
1.  can we use a function function instead of the outer anon function?  (or not, not sure about matlab compat)
2. naming it "make_1d_cell_copy"?
3. better word than `init`?  maybe `A` or `X`?

Reply:
What we are doing here is basically creating a cell array of length `n` with `init` as initial value. We need this because `cell(...)` doesn't provide a way for specifying initial value.

For 1, I haven't used nested functions before but I seldom see them being mentioned. Perhaps there're some compat issue / other drawbacks?

For 2 & 3, sure I can change that and add a comment to document the function as well (similar to python docstring).

```
  # Construct a 1D cell array of length N with initial value INIT
  make_1d_cell = @(n, init) cellfun (@(unused) init, cell (1, n), ...
                                     'UniformOutput', false);
```

or

```
  # Construct a 1D cell array of length N with initial value X
  make_1d_cell_copy = @(n, x) cellfun (@(unused) x, cell (1, n), ...
                                       'UniformOutput', false);
```

Reply:
Or how about just repeat the cellfun part:
```
%% construct 1-d cell array of length m and concatenate horizontally
MM = cellfun (@(unused) A, cell (1, m), 'UniformOutput', false);
M = horzcat (MM{:});
%% now duplicate that result vertically n times
NN = cellfun (@(unused) A, cell (1, n), 'UniformOutput', false);
B = vertcat (NN{:});
```
(not tested but you get the idea)

Reply:
maybe without `unused` var?
```
MM = cellfun (@() A, cell (1, m), 'UniformOutput', false);
```

Reply:
`@() ...` doesn't work because we need a function that takes >= 1 arguments. How about using `varargin`? I've also changed the comment a bit but the idea should be the same.

```
  %% duplicate A horizontally m times
  MM = cellfun (@(varargin) A, cell (1, m), 'UniformOutput', false);
  M = horzcat (MM{:});

  %% now duplicate that result vertically n times
  NN = cellfun (@(varargin) M, cell (1, n), 'UniformOutput', false);
  N = vertcat (NN{:});
```

Reply:
sure, `@()` seemed ok when I tried it:
```octave:1> A = [1 2 3; 4 5 6]; A = sym(A)
Symbolic pkg v3.0.1+: Python communication link active, SymPy v1.10.1.
A = (sym 2×3 matrix)
  ⎡1  2  3⎤
  ⎢       ⎥
  ⎣4  5  6⎦
octave:2> MM = cellfun (@() A, cell (1, 3), 'UniformOutput', false)
MM =
{
  [1,1] =
    <class sym>
  [1,2] =
    <class sym>
  [1,3] =
    <class sym>
}

octave:3> horzcat (MM{:})
ans = (sym 2×9 matrix)
  ⎡1  2  3  1  2  3  1  2  3⎤
  ⎢                         ⎥
  ⎣4  5  6  4  5  6  4  5  6⎦
```


But maybe its not portable (that was on Octave 6.4.0).  Anyway, `varargin` is fine

Reply:
Yep, I think it's not portable. On Octave 7.2.0, I get

```
A = (sym 2×3 matrix)

  ⎡1  2  3⎤
  ⎢       ⎥
  ⎣4  5  6⎦

octave:2> MM = cellfun (@() A, cell (1, 3), 'UniformOutput', false)
error: @<anonymous>: function called with too many inputs
error: called from
    @<anonymous>
```

This is also how it's usually handled in other languages so it shouldn't be a bug...